flink-connectors/flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/HBaseConnectorITCase.java

```diff
@@ -270,7 +282,13 @@ public void testTableSink() throws Exception {
                                 + "FROM "
                                 + TEST_TABLE_2
                                 + " AS h");
-        List<Row> results = CollectionUtil.iteratorToList(table.execute().collect());
+
+        TableResult tableResult2 = table.execute();
+        // wait to finish
+        tableResult2.await();
+
+        List<Row> results = CollectionUtil.iteratorToList(tableResult2.collect());
```

Comment:
It would look weird, if the select query has been running in batch mode and only partial result will be returned after `await()` is finished. I will do more research and come back with further information.

Reply:
Found a new way to wait the query to be finished. Please take a look.